R/run_app.R

```diff
@@ -2,15 +2,28 @@
 #'
 #' @inheritParams shared-params
 #' @export
-run_app <- function(snapshot = NULL) {
-  if (is.null(snapshot)) {
-    snapshot <- gsmApp::read_snapshot()
-  }
+
+run_app <- function(
+    dfResults = NULL,
+    dfGroups = NULL,
+    dfMetrics = NULL,
+    dfBounds = NULL) {
+  dfResults <- dfResults %||% gsm::reportingResults %>%
+    dplyr::filter(.data$SnapshotDate == max(.data$SnapshotDate))
+  dfGroups <- dfGroups %||% gsm::reportingGroups
+  dfMetrics <- dfMetrics %||% gsm::reportingMetrics
+  dfBounds <- dfBounds %||% gsm::reportingBounds %>%
+    dplyr::filter(.data$SnapshotDate == max(.data$SnapshotDate))
 
   shinyApp(
     ui = ui(),
     server = function(input, output, session) {
-      gsmApp::server(input, output, session, snapshot)
+      gsmApp::server(input, output, session,
+        dfResults = dfResults,
+        dfGroups = dfGroups,
+        dfMetrics = dfMetrics,
+        dfBounds = dfBounds
+      )
     }
```

Comment:
Our function should take `dfResults`, `dfGroups`, `dfMetrics`, and `dfBounds`, and return a `server` function that takes `input, output, session` as args. That will make the calling here more straightforward/easier to understand.

I'm going to separate (most? all?) of these things about "how I wish things were structured" out into issues so we can discuss them there without blocking this PR from being merged.

Reply:
Split into #149.